h2o-core/src/main/java/water/fvec/Frame.java

```diff
@@ -95,12 +95,14 @@ public Frame(Key<Frame> key, Vec vecs[], boolean noChecks) {
     super(key);
     assert noChecks;
     _vecs = vecs;
-    _names = new String[vecs.length];
+    String[] names = new String[vecs.length];
     _keys = makeVecKeys(vecs.length);
     for (int i = 0; i < vecs.length; i++) {
-      _names[i] = defaultColName(i);
+      names[i] = defaultColName(i);
       _keys[i] = vecs[i]._key;
     }
+    
+    setNames(names);
```

Comment:
It is dangerous to call non-final public method in a Constructor, this method can be overridden in a derived class & it might access to fields that are not yet initialized